src/main/java/org/prebid/server/bidder/imds/ImdsBidder.java

```diff
@@ -75,19 +80,31 @@ public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequ
 
         final BidRequest outgoingRequest = bidRequest.toBuilder()
                 .imp(validImps)
-                .ext(mapper.fillExtension(ExtRequest.empty(), ExtRequestSynacormedia.of(firstExtImp.getSeatId())))
+                .ext(mapper.fillExtension(ExtRequest.empty(), ExtRequestImds.of(firstExtImp.getSeatId())))
                 .build();
 
-        return Result.of(Collections.singletonList(
+        return Result.of(
+                Collections.singletonList(
                         BidderUtil.defaultRequest(
-                                outgoingRequest,
-                                endpointUrl.replaceAll("\\{\\{Host}}", firstExtImp.getSeatId()),
-                                mapper)),
-                errors);
+                            outgoingRequest,
+                            generateEndpointUrl(firstExtImp),
+                            mapper
+                    )
+                ),
+                errors
+        );
     }
 
-    private ExtImpSynacormedia parseAndValidateExtImp(ObjectNode impExt) {
-        final ExtImpSynacormedia extImp = parseExtImp(impExt);
+    private String generateEndpointUrl(ExtImpImds firstExtImp) {
+        String accountId = URLEncoder.encode(firstExtImp.getSeatId(), StandardCharsets.UTF_8);
+        String sourceId = URLEncoder.encode(prebidVersion, StandardCharsets.UTF_8);
```

Comment:
please, use `final` keyword everywhere where variable is not reassigned. Also, please, check for similar occurences.